APMrover2/RC_Channel.cpp

```diff
@@ -161,6 +164,20 @@ void RC_Channel_Rover::do_aux_function(const aux_func_t ch_option, const aux_swi
         }
         break;
 
+    case AVOID_PROXIMITY:
```

Comment:
This should not be here.

It should be in `RC_Channel/RC_Channel.cpp` so Copter and Sub can also take advantage of the option.  This will require a singleton be created for `AC_Avoid`.

Reply:
Yes, having a singleton would be nice.  It crossed my mind but Rover doesn't log events so I gave up on the idea.  If you've got some ideas around that it would be nice to have more shared code.